pkg/controller/disruption/disruption.go

```diff
@@ -173,94 +175,84 @@ func (dc *DisruptionController) finders() []podControllerFinder {
 		dc.getPodStatefulSets}
 }
 
+var (
+	controllerKindRS  = v1beta1.SchemeGroupVersion.WithKind("ReplicaSet")
+	controllerKindSS  = apps.SchemeGroupVersion.WithKind("StatefulSet")
+	controllerKindRC  = v1.SchemeGroupVersion.WithKind("ReplicationController")
+	controllerKindDep = v1beta1.SchemeGroupVersion.WithKind("Deployment")
+)
+
 // getPodReplicaSets finds replicasets which have no matching deployments.
 func (dc *DisruptionController) getPodReplicaSets(pod *v1.Pod) ([]controllerAndScale, error) {
-	cas := []controllerAndScale{}
-	rss, err := dc.rsLister.GetPodReplicaSets(pod)
-	// GetPodReplicaSets returns an error only if no ReplicaSets are found.  We
-	// don't return that as an error to the caller.
-	if err != nil {
-		return cas, nil
-	}
-	controllerScale := map[types.UID]int32{}
-	for _, rs := range rss {
-		// GetDeploymentsForReplicaSet returns an error only if no matching
-		// deployments are found.
-		_, err := dc.dLister.GetDeploymentsForReplicaSet(rs)
-		if err == nil { // A deployment was found, so this finder will not count this RS.
-			continue
+	var casSlice []controllerAndScale
+	if controllerRef := controller.GetControllerOf(pod); controllerRef != nil {
+		if controllerRef.Kind == controllerKindRS.Kind {
+			rs, err := dc.rsLister.ReplicaSets(pod.Namespace).Get(controllerRef.Name)
+			if err != nil {
+				return nil, nil
+			}
+			controllerRef := controller.GetControllerOf(rs)
```

Comment:
After fetching the RS by name, check that the UID matches `controllerRef.UID`. If it doesn't match, the RS you fetched doesn't actually own the pod; it just has the same name as a previously-deleted RS that owned the pod. In that case, you can `return nil, nil` since the pod has no RS owner.